presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/TransformUncorrelatedInPredicateSubqueryToSemiJoin.java

```diff
@@ -249,20 +249,22 @@ else if (node.getSource().getOutputSymbols().contains(symbol)) {
         @Override
         public PlanNode visitApply(ApplyNode node, RewriteContext<Void> context)
         {
-            Symbol value = Symbol.from(inPredicate.getValue());
-            Symbol valueList = Symbol.from(inPredicate.getValueList());
-            if (node.getCorrelation().isEmpty() && inPredicateMatchesApply(node, value, valueList)) {
-                checkState(!semiJoinSymbol.isPresent(), "Semi join symbol is already set");
-                semiJoinSymbol = Optional.of(symbolAllocator.newSymbol("semijoin_result", BOOLEAN));
-                return new SemiJoinNode(idAllocator.getNextId(),
-                        node.getInput(),
-                        node.getSubquery(),
-                        value,
-                        valueList,
-                        semiJoinSymbol.get(),
-                        Optional.empty(),
-                        Optional.empty()
-                );
+            if (node.getCorrelation().isEmpty()) {
```

Comment:
So far when there is correlation `StatementAnalyzer` raises exception, so I didn't have to check if node is correlated. 

One of the previous (in this PR) commit changes that and updates correlation and so this rewrite has to check if apply is correlated.

I moved this commit before the commit which updates correlation.